pynest/nest/lib/hl_api_spatial.py

```diff
@@ -810,6 +888,87 @@ def GetTargetPositions(sources, tgt_layer, syn_model=None):
     return [src_tgt_pos_map[snode_id] for snode_id in sources.tolist()]
 
 
+def GetSourcePositions(src_layer, targets, syn_model=None):
+    """
+    Obtain positions of sources to a given `NodeCollection` of `targets`.
+
+    For each neuron in `targets`, this function finds all source elements
+    in `src_layer`. If `syn_model` is not given (default), all targets are
+    returned, otherwise only source of specific type.
+
+    Parameters
+    ----------
+    src_layer : NodeCollection
+        `NodeCollection` of src_layer
+    targets : NodeCollection
+        `NodeCollection` with node ID(s) of target neurons
+    syn_type : [None | str], optional, default: None
+        Return only source positions for a given synapse model.
+
+    Returns
+    -------
+    list of list(s) of tuple(s) of floats:
+        Positions of source neurons fulfilling the given criteria as a nested
+        list, containing one list of positions per node in targets.
+
+    See also
+    --------
+    GetSourceNodes: Obtain sources of a `NodeCollection` of targets in a given source
+    population.
+
+    Notes
+    -----
+    * For distributed simulations, this function only returns sources on the
+      local MPI process.
+
+    Example
+    -------
+        ::
+
+            import nest
+
+            # create a spatial population
+            s_nodes = nest.Create('iaf_psc_alpha', positions=nest.spatial.grid(shape=[11, 11], extent=[11., 11.]))
+
+            # connectivity specifications with a mask
+            conndict = {'rule': 'pairwise_bernoulli', 'p': 1.,
+                        'mask': {'rectangular': {'lower_left' : [-2.0, -1.0],
+                                                 'upper_right': [2.0, 1.0]}}}
+
+            # connect population s_nodes with itself according to the given
+            # specifications
+            nest.Connect(s_nodes, s_nodes, conndict)
+
+            # get the positions of the targets of a source neuron
+            nest.GetSourcePositions(s_nodes, s_nodes[5])
+    """
+    if not isinstance(targets, NodeCollection):
```

Comment:
Empty line above

Reply:
@heplesser Since you have approved, can this conversation be resolved?